### PR TITLE
Add new column component

### PR DESCRIPTION
**Pull request by kei95**
# ~~⚠️Please merge https://github.com/ryonryon/trello-clone/pull/67 first⚠️~~

### Ticket
https://github.com/ryonryon/trello-clone/issues/60

### Description
- Add add column endpoint to add new column to the project
- Add `<AdditionColumn />` component which lets the user to add new columns

### CODE REVIEW

File: src/components/AdditionColumn/AdditionColumn.test.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  });
+
+  it("Press on the column - It should expand the component and show more items", async () => {
+    const closedColumnText = "Add another list";
+    const addButtonText = "Add card";
+    const editableTitlePlacehold = "Enter list title...";
+
+    // before press on the component - it should show default text
+    expect(screen.getByText(closedColumnText)).toBeTruthy();
+    expect(screen.getByTestId("AdditionColumn_expandableContainer")).toHaveStyleRule("opacity: 0");
```
Comment (kei95) [inline src/components/AdditionColumn/AdditionColumn.test.tsx:30]: if you wanted to check component's style, use this `toHaveStyleRule` function. In order to do so, make sure to import `"jest-styled-components"`

File: src/components/AdditionColumn/AdditionColumn.tsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    if (isLoading) return;
+    const newColumn = await requestCreateColumn({ variables: { name: columnTitleValue } });
+
+    if (newColumn) {
+      const mappedNewColumn: Column = {
+        id: newColumn.id,
+        name: newColumn.name,
+        order: newColumn.sort,
+        tickets: [],
+      };
```
Comment (kei95) [inline src/components/AdditionColumn/AdditionColumn.tsx:55]: These lines probably can go under `api` or `util` as we might see this pattern often

File: src/components/AdditionColumn/AdditionColumn.test.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
+import { act, fireEvent, render, screen } from "@testing-library/react";
+import { TestRendererWithContext } from "../../utils/testRendererWithContext";
+
+import AdditionColumn from "./AdditionColumn";
```
Comment (ryonryon) [inline src/components/AdditionColumn/AdditionColumn.test.tsx]:
```suggestion
import { act, fireEvent, render, screen } from "@testing-library/react";

import { TestRendererWithContext } from "../../utils/testRendererWithContext";
import AdditionColumn from "./AdditionColumn";
```

File: src/components/AdditionColumn/AdditionColumn.test.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+beforeEach(() => {
+  act(() => {
+    render(<AdditionColumn />, {
+      wrapper: TestRendererWithContext,
+    });
+  });
+});
+
+describe("<AdditionColumn />", () => {
+  it("Default render - It should render closed state of the component", () => {
```
Comment (ryonryon) [inline src/components/AdditionColumn/AdditionColumn.test.tsx]: It is a very small thing and you don't need to change. 
I think it is better to write it("should render closed state of the component", (() => {}) when you use it.

File: src/components/AdditionColumn/AdditionColumn.test.tsx (hunk 3 above)
Comment (kei95) [reply src/components/AdditionColumn/AdditionColumn.test.tsx]: https://github.com/ryonryon/trello-clone/pull/68/commits/e8367cf9b3c4f1eebea44e89a35d3b290bf0c01a

File: src/components/AdditionColumn/AdditionColumn.test.tsx (hunk 4 above)
Comment (kei95) [reply src/components/AdditionColumn/AdditionColumn.test.tsx]: I don't quite understand what you mean, could you elaborate on that a bit further? (Don't text me in LINE with Japanese for this BTW😂)

**Review by ryonryon — APPROVED** [review]

File: src/components/AdditionColumn/AdditionColumn.test.tsx (hunk 4 above)
Comment (ryonryon) [reply src/components/AdditionColumn/AdditionColumn.test.tsx]: if you write test with test function.
`test("passed projectId - it should be rendered correctly, () => {})` 
> test passed projectId - it should be rendered correctly.

if you write test with it function
`it("Default render - It should render closed state of the component", () => {})` 
> it Default render - It should closed state of the component. 

As english sentence it is not grammarly correct. so It should be something like
`it("should render closed state of the component as default render", () => {})` 
> it should closed state of the component as default render.

Comment (kei95) [reply src/components/AdditionColumn/AdditionColumn.test.tsx]: Ah makes sense! Yeah I mixed up with `test` and `it` honestly - I'll change `it` to `test` in this file. Thanks for the advice👍

Comment (kei95) [reply src/components/AdditionColumn/AdditionColumn.test.tsx]: https://github.com/ryonryon/trello-clone/pull/68/commits/610d79486ca4e694599c28c1d1bde01269f6bb25

**Pull request merged**